studio/components/interfaces/TableGridEditor/TableGridEditor.tsx

```diff
@@ -189,18 +197,19 @@ const TableGridEditor = ({
     }
   }, [selectedTable?.id])
 
+  const entityType = useEntityType(selectedTable?.id)
+
   // NOTE: DO NOT PUT HOOKS AFTER THIS LINE
   if (isUndefined(selectedTable)) {
     return <NotFoundState id={Number(id)} />
   }
 
   const tableId = selectedTable?.id
 
-  // @ts-ignore
-  const schema = meta.schemas.list().find((schema) => schema.name === selectedSchema)
-  const isViewSelected = !Object.keys(selectedTable).includes('rls_enabled')
-  const isForeignTableSelected = meta.foreignTables.byId(selectedTable.id) !== undefined
-  const isLocked = meta.excludedSchemas.includes(schema?.name ?? '')
+  const isViewSelected = entityType?.type === ENTITY_TYPE.VIEW
+  const isTableSelected = entityType?.type === ENTITY_TYPE.TABLE
+  const isForeignTableSelected = entityType?.type === ENTITY_TYPE.FOREIGN_TABLE
+  const isLocked = meta.excludedSchemas.includes(entityType?.schema ?? '')
   const canUpdateTables = checkPermissions(PermissionAction.TENANT_SQL_ADMIN_WRITE, 'tables')
   const canEditViaTableEditor = !isViewSelected && !isForeignTableSelected && !isLocked
```

Comment:
think we can change it to this instead? since we only allow edit functionality against tables, not any other entity types

```suggestion
  const canEditViaTableEditor = isTableSelected && !isLocked
```